server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/DeleteConditionAction.java

```diff
@@ -62,9 +61,14 @@ public void define(WebService.NewController controller) {
   public void handle(Request request, Response response) {
     long conditionId = request.mandatoryParamAsLong(PARAM_ID);
     try (DbSession dbSession = dbClient.openSession(false)) {
-      OrganizationDto organization = wsSupport.getOrganization(dbSession);
-      userSession.checkPermission(ADMINISTER_QUALITY_GATES, organization);
-      dbClient.gateConditionDao().delete(wsSupport.getCondition(dbSession, conditionId), dbSession);
+
+      QualityGateConditionDto condition = wsSupport.getCondition(dbSession, conditionId);
+      QualityGateDto qualityGateDto = dbClient.qualityGateDao().selectById(dbSession, condition.getQualityGateId());
+      String message = format("Condition '%s' is linked to an unknown quality gate '%s'", conditionId, condition.getQualityGateId());
```

Comment:
I think this message field can be removed and used directly in `checkState`

Reply:
Indeed, I'll have a look